specs/CustomDownload.md

```diff
@@ -0,0 +1,775 @@
+            CHECK_FAILURE(download->get_State(&downloadState));
+            switch (downloadState)
+            {
+            case DOWNLOAD_IN_PROGRESS:
+                break;
+            case DOWNLOAD_INTERRUPTED:
+                // Here developer can take different actions based on `args->InterruptReason`.
+                MessageBox(nullptr, L"Connection with file host was broken.", nullptr, MB_OK);
+                CHECK_FAILURE(download->get_CanResume(&canResume));
+                if (canResume)
+                {
+                    download->Resume();
+                }
+                else
+                {
+                    CompleteDownload(download);
+                }
+                break;
+            case DOWNLOAD_COMPLETE:
+                CompleteDownload(download);
+                break;
+            }
+            return S_OK;
+        })
+    .Get(),
+&m_downloadStateChangedToken));
+```
+## .NET/ WinRT
+```c#
+webView.CoreWebView2.DownloadStarting += delegate (object sender, CoreWebView2DownloadStartingEventArgs args)
+{
+    CoreWebView2Deferral deferral = args.GetDeferral();
+    System.Threading.SynchronizationContext.Current.Post((_) =>
+    {
+        using (deferral)
+        {
+            if (_coreWebView2Settings == null)
+            {
+                _coreWebView2Settings = webView.CoreWebView2.Settings;
+            }
+            _coreWebView2Settings.ShouldDisplayDefaultDownloadDialog = false;
+
+            var dialog = new TextInputDialog(
+                title: "Download Starting",
+                description: "Enter new result file path or select OK to keep default path. Select cancel to cancel the download.",
+                defaultInput: args.ResultFilePath);
+            if (dialog.ShowDialog() == true)
+            {
+                args.ResultFilePath = dialog.Input.Text;
+                UpdateProgress(args.Download());
+            }
+            else
+            {
+                args.Cancel = true;
+            }
+        }
+    }, null);
+};
+
+// Update download progress
+void UpdateProgress(CoreWebView2Download download)
+{
+    download.DownloadProgressSizeInBytesChanged += delegate (object sender, Object e)
+    {
+        // Here developer can update download dialog to show progress of a download using
+        // `download.DownloadProgressSizeInBytes` and `download.DownloadSizeInBytes`
+    };
+
+    download.DownloadStateChanged += delegate (object sender, CoreWebView2DownloadStateChangedEventArgs args)
+    {
+        switch (download.State)
+        {
+        case CoreWebView2DownloadState.DownloadInProgress:
+            break;
+        case CoreWebView2DownloadState.DownloadInterrupted:
+            // Here developer can take different actions based on `args.InterruptReason`.
+            MessageBox.Show(this, "Connecton with file host broken.", "Download Interrupted");
+            if (download.CanResume == true)
+            {
+                download.Resume();
```

Comment:
We probably don't want to auto resume. If we're interrupted for a reason that must be resolved and we never try to resolve it or tell the user who can try to resolve, then we're just endlessly getting interrupted & resuming. We probably don't want to tell end devs to do it this way.